### PR TITLE
Update: Implement auto fix for space-unary-ops rule (fixes #3976)

### LINKED ISSUE #3976: Implement auto fix functionality for space-unary-ops rule
I can work on this one

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/27043688-implement-auto-fix-functionality-for-space-unary-ops-rule?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by alberto**

### CODE REVIEW

File: lib/rules/space-unary-ops.js [hunk 1]
```diff
@@ -17,12 +17,13 @@ module.exports = function(context) {
...
     /**
-    * Check if the parent unary operator is "!" in order to know if it's "!!" convert to Boolean or just "!" negation
+    * Check if the node is the first "!" in a "!!" convert to Boolean expression
     * @param {ASTnode} node AST node
-    * @returns {boolean} Whether or not the parent is unary "!" operator
+    * @returns {boolean} Whether or not the node is first "!" in "!!"
     */
-    function isParentUnaryBangExpression(node) {
-        return node && node.parent && node.parent.type === "UnaryExpression" && node.parent.operator === "!";
+    function isFirstBangInBangBangExpression(node) {
```
Comment (ilyavolodin) [inline lib/rules/space-unary-ops.js]: This looks like unrelated fix for some other issue. Is that correct?

Comment (alberto) [inline lib/rules/space-unary-ops.js]: Oh, yes. It was reporting on the first bang instead of the second.

Comment (ilyavolodin) [inline lib/rules/space-unary-ops.js]: Can you update unittests and add column and line numbers of the errors for this fix then?

Comment (alberto) [inline lib/rules/space-unary-ops.js]: Done

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**